Modules/mathmodule.c

```diff
@@ -2016,6 +2016,71 @@ math_factorial(PyObject *module, PyObject *arg)
 }
 
 
+/*[clinic input]
+math.lcm
+
+    x as a: object
+    y as b: object
+    /
+least common multiple of x and y
+[clinic start generated code]*/
+
+static PyObject *
+math_lcm_impl(PyObject *module, PyObject *a, PyObject *b)
+/*[clinic end generated code: output=6f83fb6d671074ba input=bd41b785dc2a4ff1]*/
+{
+    PyObject *g, *m, *f, *ab;
+    
+    a = PyNumber_Index(a);
+        if (a == NULL) {
+            return NULL;
+    }
+    b = PyNumber_Index(b);
```

Comment:
At this point, the previous `PyNumber_Index` call didn't fail, so we definitely own a reference to `a`.
The call to `PyNumber_Index` in _this_ line returns us a new reference, or `None`.

Owned references before this line: `a`
Owned references after this line: `a`, `b`, unless the call failed, in which case just `a`.